### PR TITLE
fixed scheduler checkpoint loading, typo

### PR DESCRIPTION
**Pull request by AnnihilatorChess**
Hello,

I found a bug where resuming a run from a checkpoint incorrectly restarts the LR scheduler's warmup and cosine decay. This is because the Trainer in training.py saves and loads the optimizer state but not the lr_scheduler state.

This PR fixes the save_model and load_checkpoint methods to include the lr_scheduler.state_dict() in the checkpoint, ensuring that training resumes with the correct learning rate.

(I also fixed a small typo: optimizer_state_dit -> optimizer_state_dict.)"

### CODE REVIEW

**Comment by kazewong** [comment]
@AnnihilatorChess This was accidentally closed during a restructuring of the repo. We would love to have your contribution, so once we are done with the restructuring and release, we will ping you for submitting a PR again.
